Add PyCharm, notebooks, egg-info to gitignore.

petterbejo opened the pull request:
Thought it might be a good idea to update the gitignore to avoid things getting messy.

WilliamHPNielsen commented on .gitignore:
```diff
@@ -1,2 +1,13 @@
+.ipynb_checkpoints/
+limnos.egg-info/
+limnos/.ipynb_checkpoints/
+limnos/Untitled.ipynb
```
Why is `Untitled.ipynb` explicitly ignored? Nothing in limnos generates such a file (right?)

WilliamHPNielsen commented on .gitignore:
```diff
@@ -1,2 +1,13 @@
+limnos/.ipynb_checkpoints/
+limnos/Untitled.ipynb
+limnos/__pycache__/
+limnos/tests/__pycache__/
```
It is sufficient to have a single `__pycache__` line; we don't need one for each subfolder.

WilliamHPNielsen commented on .gitignore:
```diff
@@ -1,2 +1,13 @@
+
+.ipynb_checkpoints/
+limnos.egg-info/
+limnos/.ipynb_checkpoints/
```
This line is redundant as line 7 already excludes checkpoints.

WilliamHPNielsen reviewed (changes requested):
I left some nitpicky inline comments that I think we should address before merging.

petterbejo commented:
I've tidied the gitignore by removing duplicate pycache and checkpoints folders. I've also removed the Untitled.ipynb file from the gitignore and renamed it Limnos-UI since I consider this being more user friendly (I could have made a separate branch for this, but since it's such a small change and it is somehow related to this one, I've included it here).

WilliamHPNielsen reviewed (approved):
Sweet!
